bugbug/models/rcatype.py

```diff
@@ -0,0 +1,186 @@
+            bug_features.is_coverity_issue(),
+            bug_features.has_crash_signature(),
+            bug_features.has_url(),
+            bug_features.has_w3c_url(),
+            bug_features.has_github_url(),
+            # Ignore whiteboards that would make the ML completely skewed
+            # bug_features.whiteboard(),
+            bug_features.patches(),
+            bug_features.landings(),
+            bug_features.blocked_bugs_number(),
+            bug_features.ever_affected(),
+            bug_features.affected_then_unaffected(),
+            bug_features.product(),
+            bug_features.component(),
+        ]
+
+        cleanup_functions = [
+            feature_cleanup.url(),
+            feature_cleanup.fileref(),
+            feature_cleanup.synonyms(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                (
+                    "bug_extractor",
+                    bug_features.BugExtractor(feature_extractors, cleanup_functions),
+                ),
+                (
+                    "union",
+                    ColumnTransformer(
+                        [
+                            ("data", DictVectorizer(), "data"),
+                            ("title", self.text_vectorizer(min_df=0.001), "title"),
+                            (
+                                "first_comment",
+                                self.text_vectorizer(min_df=0.001),
+                                "first_comment",
+                            ),
+                            (
+                                "comments",
+                                self.text_vectorizer(min_df=0.001),
+                                "comments",
+                            ),
+                        ]
+                    ),
+                ),
+            ]
+        )
+
+        self.clf = OneVsRestClassifier(xgboost.XGBClassifier(n_jobs=16))
+
+    # return rca from a whiteboard string
+    def get_rca_from_whiteboard(self, whiteboard_data):
+        rca = []
+        whiteboard_data = re.sub(" ", "", whiteboard_data).lower()
+        for whiteboard in whiteboard_data.split("["):
+            if not any(whiteboard.startswith(s) for s in ("rca-", "rca:")):
+                continue
+
+            rca_whiteboard = re.sub("]", "", whiteboard)
+
+            # Hybrid cases: rca:X-Y
+            rca_whiteboard = re.sub(":", "-", rca_whiteboard)
+
+            rca_whiteboard_split = (
+                rca_whiteboard.split("-", 1)
+                if self.rca_subcategories_enabled
+                else rca_whiteboard.split("-")
+            )
+
+            if rca_whiteboard_split[1] not in self.RCA_LIST:
+                logger.warning(rca_whiteboard_split[1] + " not in RCA_LIST")
+            else:
+                rca.append(rca_whiteboard_split[1])
+        return rca
+
+    def get_labels(self):
+        # return rca for all bugs
+        def get_rca_from_bugs(bugs):
```

Comment:
Nit: I don't see the need for this function, why not making it inline directly in the outer function (`get_labels`)?